tests-e2e/cypress/integration/runs/rdp_main_checklist_spec.js

```diff
@@ -43,18 +71,77 @@ describe('runs > run details page > checklist', () => {
             playbookRunName: 'the run name',
             ownerUserId: testUser.id,
         }).then((playbookRun) => {
+            testRun = playbookRun;
+
             // # Visit the playbook run
             cy.visit(`/playbooks/run_details/${playbookRun.id}`);
         });
     });
 
-    it('is visible', () => {
-        // * Verify the tasks section is present
-        cy.findByTestId('run-checklist-section').should('be.visible');
+    const getChecklist = () => cy.findByTestId('run-checklist-section');
+    const getChecklistTasks = () => getChecklist().findAllByTestId('checkbox-item-container');
+
+    const commonTests = () => {
+        it('is visible', () => {
+            // * Verify the tasks section is present
+            getChecklist().should('be.visible');
+        });
+
+        it('has title', () => {
+            // * Verify the task section has a title
+            getChecklist().find('h3').contains('Tasks');
+        });
+
+        it('can see the tasks', () => {
+            // * Verify tasks are shown
+            getChecklistTasks().should('have.length', 4);
+        });
+    };
+
+    describe('as participant', () => {
+        commonTests();
+
+        it('click marks task as done', () => {
+            // # Click first task
+            getChecklistTasks().eq(taskIndex).find('.checkbox').check();
+
+            // *  Assert checkbox is checked
+            getChecklistTasks().eq(taskIndex).find('.checkbox').should('be.checked');
+        });
+
+        it('has hover menu', () => {
+            // # Hover over the checklist item
+            getChecklistTasks().eq(taskIndex).trigger('mouseover');
+
+            // # Click dot menu
+            getChecklistTasks().eq(taskIndex).findByTitle('More').click();
+
+            // * Assert actions are available
+            cy.findByRole('button', {name: 'Skip task'}).should('be.visible');
+            cy.findByRole('button', {name: 'Duplicate task'}).should('be.visible');
+        });
     });
 
-    it('has title', () => {
-        // * Verify the task section has a title
-        cy.findByTestId('run-checklist-section').find('h3').contains('Tasks');
+    describe('as viewer', () => {
+        beforeEach(() => {
+            cy.apiLogin(testViewerUser).then(() => {
+                cy.visit(`/playbooks/run_details/${testRun.id}`);
+            });
+        });
+
+        commonTests();
+
+        it('click does not work', () => {
+            // # Click first task
+            getChecklistTasks().eq(taskIndex).find('.checkbox').should('be.disabled');
+        });
+
+        it('has not hover menu', () => {
+            // # Hover over the checklist item
+            getChecklistTasks().eq(taskIndex).trigger('mouseover');
+
+            // # Click dot menu
```

Comment:
Copypaste typo, I think
```suggestion
            // * Check that the hover menu is not rendered
```